### PR TITLE
user general information tab

### LINKED ISSUE #2660: General information tab
https://www.figma.com/proto/XS4Ag6GWcgdq2dBlLzBkay/Admin-v3?node-id=429%3A2292&scaling=min-zoom&page-id=0%3A1&starting-point-node-id=410%3A1984&hide-ui=1

Screen Shot 2022-05-04 at 11 40 00 AM

The Notes data is specific to a PoolCandidate, not a User. We should list a set of Note inputs, one for each Pool the user is in.

## Acceptance Criteria
- [ ] Each interaction (change status, remove from pool, add to pool, save notes) may need its own mutation
- [ ]

### PR DESCRIPTION
**Pull request by JamesHuf**
Resolves #2660.

Fills out the the general information tab on the candidate details page.

Notes:
- ~~The add user to pool functionality still has some issues I need to address.~~
- ~~I should probably add a story for the tab but I haven't yet.~~

### CODE REVIEW

File: frontend/admin/src/js/components/user/GeneralInfoTabDialogs.tsx
Comment (esizer) [inline frontend/admin/src/js/components/user/GeneralInfoTabDialogs.tsx:635]: Since we can't add a user to a pool they are already in, should we filter those out of the available options, or disable them?

File: frontend/admin/src/js/components/user/GeneralInfoTabDialogs.tsx
Comment (esizer) [inline frontend/admin/src/js/components/user/GeneralInfoTabDialogs.tsx:52]: I'm not sure if we are doing this elsewhere but it may be a good idea to disable the submit button while a mutation is running to prevent multiple submissions at once.

File: frontend/admin/src/js/components/user/GeneralInformationTab.tsx
Comment (esizer) [inline frontend/admin/src/js/components/user/GeneralInformationTab.tsx:637]: You can actually just pass the icon directly to the `<TableofContents.Heading>` component as a prop and it will apply styles for you.

```tsx
<TableOfContents.Heading icon={UserIcon}>
  {item.title}
</TableOfContents.Heading>
```

**Review by esizer — CHANGES_REQUESTED** [review]
Very nice! Everything is working great, just a few minor suggestions.

File: frontend/admin/src/js/components/user/GeneralInfoTabDialogs.tsx
Comment (JamesHuf) [reply frontend/admin/src/js/components/user/GeneralInfoTabDialogs.tsx:635]: Yeah, it's on my list.

**Comment by vd1992** [comment]
that's a real meaty PR!

File: frontend/admin/src/js/components/user/GeneralInformationTab.tsx
Comment (esizer) [inline frontend/admin/src/js/components/user/GeneralInformationTab.tsx]: Other sections have this. May be a good idea to add it here as well for consistency.

```suggestion
      data-h2-radius="b(s)"
      data-h2-font-size="b(caption)"
```

**Review by esizer — CHANGES_REQUESTED** [review]
Everything is looking great! One tiny suggestion about font size for the EE section.

File: frontend/admin/src/js/components/user/GeneralInformationTab.tsx
Comment (JamesHuf) [reply frontend/admin/src/js/components/user/GeneralInformationTab.tsx]: @Jerryescandon Could you please confirm the font sizes for the various sections are as intended when you review the UI?

Comment (JamesHuf) [reply frontend/admin/src/js/components/user/GeneralInformationTab.tsx]: I'll try to bring the heading styles more in line but I was actually wondering about the difference between the text here:
image
and here:
image
image

Comment (JamesHuf) [reply frontend/admin/src/js/components/user/GeneralInformationTab.tsx]: Am I right in thinking those heading styles are a general rule for the whole site (or just admin)? If so it might be good idea to create a new issue for bringing everything more in line with this.

**Comment by petertgiles** [comment]
> Managers should only see notes for pools they have access to.

I guess since we don't have a manager role yet this AC is out of scope?

File: frontend/admin/src/js/components/user/GeneralInformationTab/GeneralInformationTab.tsx
Comment (petertgiles) [inline frontend/admin/src/js/components/user/GeneralInformationTab/GeneralInformationTab.tsx:412]: Please use a parameterized formatMessge instead of a concat.  It's better to give the translator the whole message to work with.

File: frontend/admin/src/js/components/user/GeneralInformationTab/GeneralInformationTab.tsx
Comment (petertgiles) [inline frontend/admin/src/js/components/user/GeneralInformationTab/GeneralInformationTab.tsx:423]: Parameterize, please.

**Review by petertgiles — CHANGES_REQUESTED** [review]
Wow, this is some top-shelf work, James.  I have a small request for translation but otherwise love it!

File: frontend/admin/src/js/components/user/GeneralInformationTab.tsx
Comment (Jerryescandon) [reply frontend/admin/src/js/components/user/GeneralInformationTab.tsx]: So the only thing I would recommend would be to increase these up to the same size as "body" used at the bottom.


image

Comment (Jerryescandon) [reply frontend/admin/src/js/components/user/GeneralInformationTab.tsx]: > Am I right in thinking those heading styles are a general rule for the whole site (or just admin)? If so it might be a good idea to create a new issue for bringing everything more in line with this.

I would think Admin for now, Im unsure where we are headed with the public facing stuff. So I would hold off on doing any cleanup on that side yet.

Comment (JamesHuf) [reply frontend/admin/src/js/components/user/GeneralInformationTab.tsx]: Sounds good, I'll make those changes for this page and create an issue for the headers for the rest of Admin.

Comment (JamesHuf) [reply frontend/admin/src/js/components/user/GeneralInformationTab.tsx]: - Headers in the component changed sizes to match recommendation
- Created #3064 to change over the rest of the admin headings
- Replaced caption size text with regular size

**Comment by petertgiles** [comment]
It looks like you were working with Jerbo on styling.  @Jerryescandon could you please review the Chromatic diffs and approve if you are OK with it?
image
image

**Review by petertgiles — APPROVED** [review]

**Review by esizer — APPROVED** [review]
Awesome! 😁

**Review by Jerryescandon — APPROVED** [review]
SHIP IT

**Comment by Jerryescandon** [comment]
> It looks like you were working with Jerbo on styling. @Jerryescandon could you please review the Chromatic diffs and approve if you are OK with it? image image

I tried approving from Chromatic and it said it was approved but wouldn't update in git.

**Comment by JamesHuf** [comment]
Looks like it worked for me. You might just need to refresh. 
image

**Pull request merged**
